docs/graphql.md

```diff
@@ -0,0 +1,139 @@
+
+## Operation Name
+
+It is common (and [recommended](https://graphql.org/learn/queries/#operation-name)) to provide an _Operation Name_ for queries. Here for example `UserWithComments`:
+
+```graphql
+query UserWithComments {
+  user {
+    id
+    name
+    comments {
+      body
+    }
+  }
+}
+```
+
+The point of these are to provide an easy way for the developers, when things go wrong, to pinpoint where exactly they did so.
+
+This name is available on the server too and serves as a great distinguishing key.
+
+Span name examples:
+- `GraphQL:UserWithComments`
+- `GraphQL:UpdateUser`
+
+### Sidenote: Multiple endpoints
+
+The Node.js implementation adds the request path to the GraphQL span names.
+
+We do not find serving multiple endpoints and using them with the same Operation Names likely enough to add it to this document.
+
+## Anonymous queries
+
+An Operation Name isn't required. When one isn't provided it's hard for us to tell apart the queries.
+
+1. Some clients generate `id`s from hashing the contents of the query (see [apollo-tooling](https://github.com/apollographql/apollo-tooling/blob/1dfd737eaf85b89b2cfb13913342e091e3c03d18/packages/apollo-codegen-core/src/compiler/visitors/generateOperationId.ts#L5)). This would split the anonymous queries into separate buckets.
+
+    A problem with this approach is that a user of the APM UI has no way to recognise queries in the transactions list before clicking through.
+
+    Using just the `id` will not reveal the true culprit since there can be variables associated with the query. Different values for the variables can lead to very different workloads and response times.
+
+2. Another approach is to simply label them `[anonymous]` or something similar.
+
+    A problem with _that_ approach is that the contents and thereby the relevant db queries and other sub-span actions that the server might do while resolving these queries may be wildly different making it hard to provide a _true_ sample waterfall.
+
+    These two examples for example will look the same for the top-level GraphQL spans but will represent significantly different workloads.
+
+    ```
+    [- anonymous graphql span --------------]
+      [- 1,000x SELECT * ---------------]
+        [- 1,000 more SELECT * -]
+
+    [- anonymous graphql span --------------]
+      [- SELECT id FROM users WHERE id=? -]
+    ```
+
+    We could consider _muting_ or ignoring all sub-spans to anonymous GraphQL queries and choose to rather show nothing than potentially wrong information.
+
+No one of these are perfect. Because the benefits of using `id`s in the worst case could be misleading anyway, we're going with option 2.
+
+To further help and nudge developers to use Operation Names for their queries, we could log a warning when coming across anonymous queries.
```

Comment:
@sqren @formgeist Could we add this instead as a warning in the UI? Do we have some sort of built-in process or system of adding helpful popup (or whatever) notes to things depending on the data?

If you have a minute, I'd also love your immediate thoughts on the whole doc. If you don't then no worries 😊

Reply:
Happy to take a call anytime - but it sounds like a helpful tooltip with the appropiate nudge and information to have users make improvements in order to better analyze their data is a great addition 👍

Reply:
I agree, having a warning would be great. We already do something similar on the trace timeline to indicate if some spans are missing (dropped).

Reply:
@mikker Feel free to open a design issue in the [APM repo](https://github.com/elastic/apm/issues/new?labels=design&template=design_issue.md) and I'll have a look at coming up with an initial proposal.

Reply:
Great, thanks!